### PR TITLE
add unit test for time partitioned file sets in map/reduce

### PR DESCRIPTION
**Pull request by anew**

### CODE REVIEW

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java [hunk 1]
```diff
@@ -37,22 +43,23 @@
 
 public class TimePartitionedFileSetTest extends AbstractDatasetTest {
 
-  static TimePartitionedFileSet fileSet;
+  static final long MINUTE = TimeUnit.MINUTES.toMillis(1);
 
-  @BeforeClass
-  public static void beforeClass() throws Exception {
+  @Before
+  public void beforeClass() throws Exception {
```
Comment (albertshau) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java]: would be better to rename method name as well

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java [hunk 2]
```diff
@@ -37,22 +43,23 @@
...
     createInstance("timePartitionedFileSet", "tpfs", FileSetProperties.builder()
       .setBasePath("testDir").build());
-    fileSet = getInstance("tpfs");
   }
 
-  @AfterClass
-  public static void afterClass() throws Exception {
+  @After
+  public void afterClass() throws Exception {
```
Comment (albertshau) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java]: would be better to rename method name as well

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    // this should have created a partition in the tpfs
+    final TimePartitionedFileSet tpfs = datasetInstantiator.getDataset(PARTITIONED);
+    txExecutorFactory.createExecutor(datasetInstantiator.getTransactionAware()).execute(
+      new TransactionExecutor.Subroutine() {
+        @Override
+        public void apply() {
+          String path = tpfs.getPartition(time);
+          Assert.assertNotNull(path);
+          Assert.assertTrue(path.contains("2015-01-15/11-15"));
+          System.err.println("Path for partition: " + path);
```
Comment (albertshau) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java]: remove println

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+    // this should have created a partition in the tpfs
+    txExecutorFactory.createExecutor(datasetInstantiator.getTransactionAware()).execute(
+      new TransactionExecutor.Subroutine() {
+        @Override
+        public void apply() {
+          String path = tpfs.getPartition(time5);
+          Assert.assertNotNull(path);
+          Assert.assertTrue(path.contains("2015-01-15/11-20"));
+          System.err.println("Path for partition 5: " + path);
```
Comment (albertshau) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java]: remove println

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithTimePartitionedFileSet.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      createDataset(OUTPUT, "table");
+
+      createDataset(PARTITIONED, "timePartitionedFileSet", FileSetProperties.builder()
+        // properties for file set
+        .setBasePath("/partitioned")
+        .setInputFormat(TextInputFormat.class)
+        .setOutputFormat(TextOutputFormat.class)
+        .setOutputProperty(TextOutputFormat.SEPERATOR, SEPARATOR)
+          // properties for partitioned hive table
+          // .setEnableExploreOnCreate(true)
```
Comment (albertshau) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithTimePartitionedFileSet.java]: remove commented properties unless we're adding them back in another PR very soon.

**Comment by albertshau** [comment]
minor comments, otherwise lgtm

File: cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/FileSet.java [hunk 6]
```diff
@@ -73,4 +73,14 @@
...
   Map<String, String> getRuntimeArguments();
+
+  /**
+   * A variant of {@link co.cask.cdap.api.data.batch.InputFormatProvider#getInputFormatConfiguration}
+   * that allows passing in the input locations (rather than using the input locations that were
+   * determined from runtime arguments).
+   *
+   * @param inputLocs the input locations to be used
+   */
+  Map<String, String> getInputFormatConfiguration(Iterable<Location> inputLocs);
```
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/FileSet.java]: Better be `Iterable<? extends Location>`.

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java (hunk 1 above)
Comment (anew) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java]: done.

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java (hunk 2 above)
Comment (anew) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/dataset2/lib/TimePartitionedFileSetTest.java]: done.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java (hunk 3 above)
Comment (anew) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java]: oops. of course.

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java (hunk 4 above)
Comment (anew) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/MapReduceWithPartitionedTest.java]: done

File: cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithTimePartitionedFileSet.java (hunk 5 above)
Comment (anew) [inline cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithTimePartitionedFileSet.java]: yes, for now we can remove them.

**Comment by anew** [comment]
@albertshau @chtyim I adressed all comments. Please take another look.

**Comment by anew** [comment]
Build ran successfully: https://builds.cask.co/browse/CDAP-DUT585-1

**Comment by albertshau** [comment]
:+1:

**Pull request merged**